addons/l10n_id_efaktur/tests/test_l10n_id_efaktur.py

```diff
@@ -277,3 +287,359 @@ def test_efaktur_consume_code(self):
+
+        # If we update both together, then resulting availalbe number is
+        # combined behaviour of changing min and max
+        self.efaktur.write({'min': "0000000000005", "max": "0000000000005"})
+        self.assertEqual(self.efaktur.available, 1)
+
+        self.efaktur.write({'min': "0000000000008", "max": "0000000000010"})
+        self.assertEqual(self.efaktur.available, 3)
+
+        # Rese
+        self.efaktur.write({'min': "0000000000001"})
+
+        # Suppose we use the range 4 times, then the next_num = 5 and availability = 6
+        for i in range(4):
+            self.efaktur.pop_number()
+
+        # Once a range is used, then we can't change the min to any number
+        with self.assertRaises(ValidationError):
+            self.efaktur.write({'min': '0000000000003'})
+
+        # We cannot change the range's max to any number less than or equal to last used numbers
+        with self.assertRaises(ValidationError):
+            self.efaktur.write({'max': '0000000000003'})
+
+        # If we change the max value to number above the last used number, default behaviour applies
+        self.efaktur.write({'max': '0000000000007'})
+        self.assertEqual(self.efaktur.available, 3)
+
+    def test_next_num(self):
+        """ Test if next_num computation is correct after altering availability.
+
+        Range: 0000000000001 - 0000000000010
+        Base availability = 10
+        """
+        self.assertEqual(self.efaktur.next_num, 1)
+
+        self.efaktur.available = 5
+        self.assertEqual(self.efaktur.next_num, 6)
+
+    def test_efaktur_delete_range(self):
+        """ Ensure that user can only delete e-Faktur ranges that are unused """
+        # Use the main range and should raise error when being deleted
+        self.efaktur.pop_number()
+
+        with self.assertRaises(UserError):
+            self.efaktur.unlink()
+
+        # If a range is unused, there should be no problem when deleting it
+        new_range = self.env['l10n_id_efaktur.efaktur.range'].create({"min": "%013d" % 11, "max": "%013d" % 20})
+        new_range.unlink()
+
+    def test_efaktur_release_last_number(self):
+        """ Test that range min/max is unaffected if we're returning the last released number of that range
+        i.e. range: 1-10, 1 is used and then released back, range still 1-10 with 10 availability
+        """
+        out_invoice = self.env["account.move"].create({
+            "move_type": "out_invoice",
+            "partner_id": self.partner_id.id,
+            "invoice_date": "2019-05-01",
+            "date": "2019-05-01",
+            "invoice_line_ids": [
+                Command.create({"name": "line1", "price_unit": 110.0, "tax_ids": self.tax_id.ids}),
+            ],
+            "l10n_id_kode_transaksi": "01",
+        })
+        out_invoice.action_post()
+        self.assertEqual(self.efaktur.available, 9)
+
+        out_invoice.button_draft()
+        out_invoice.button_cancel()
```

Comment:
button_cancel already does button draft for you, so you can remove the manual call to draft.